aws-s3-source/src/main/java/dev/knative/eventing/KnativeClientOptions.java

```diff
@@ -0,0 +1,60 @@
+package dev.knative.eventing;
+
+import io.quarkus.tls.runtime.keystores.TrustAllOptions;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PfxOptions;
+import io.vertx.ext.web.client.WebClientOptions;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Named;
+import org.apache.camel.CamelContext;
+import org.apache.camel.spi.PropertiesComponent;
+
+/**
+ * Knative client options able to configure secure Http transport options.
+ * SSL options are added automatically when enabled via
+ */
+@ApplicationScoped
+public class KnativeClientOptions {
+
+    private static final String PROPERTY_PREFIX = "camel.knative.client.ssl.";
+
+    @Named("knativeClientOptions")
+    public WebClientOptions knativeClientOptions(CamelContext camelContext) {
+        PropertiesComponent propertiesComponent = camelContext.getPropertiesComponent();
+
+        boolean sslEnabled = Boolean.parseBoolean(
+                propertiesComponent.resolveProperty(PROPERTY_PREFIX + "enabled").orElse("false"));
+
+        WebClientOptions options = new WebClientOptions();
+        if (sslEnabled) {
+            options.setSsl(true);
+
+            boolean verifyHostname = Boolean.parseBoolean(
+                    propertiesComponent.resolveProperty(PROPERTY_PREFIX + "verify.hostname").orElse("false"));
+            options.setVerifyHost(verifyHostname);
```

Comment:
I would make this `true` by default or even disallow disabling host verification
```suggestion
            boolean verifyHostname = Boolean.parseBoolean(
                    propertiesComponent.resolveProperty(PROPERTY_PREFIX + "verify.hostname").orElse("true"));
            options.setVerifyHost(verifyHostname);
```

Reply:
at least for integration tests we need to disable. but `true` seems to be a good default value, thx